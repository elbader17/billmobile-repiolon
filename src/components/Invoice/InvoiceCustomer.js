import React from 'react';
import { View, TextInput, ScrollView } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { Button } from "react-native-elements";
import PropTypes from 'prop-types';
import InvoiceListCustomer from './InvoiceListCustomers';
import { validateDni } from '../../utils/validations'
import style from './style';

const InvoiceCustomer = props => {
  const iconEnabled = <Icon name="md-return-right" size={18} color="#EE6123"/>;
  const iconDisabled = <Icon name="md-return-right" size={18} color="#858585"/>;
  if (!props.finalConsumer) {
    return (
      <ScrollView>
        <View style={style.listCustomer}>
          <InvoiceListCustomer
            identityCustomer={props.fiscalIdentity}
          />
        </View>
      </ScrollView>
    );
  }else {
    return (
      <View style={style.containerFinalConsumer}>
        <View style={[style.inLineSpaceBetween, {alignItems: 'center'}]}>
          <TextInput
            placeholder="DNI"
            placeholderTextColor={'#cecece'}
            onChangeText={props.setFinalConsumer}
            style={[style.textRegular18GrayDark,style.inputDNICustomer]}
            keyboardType='numeric'
          />
          <Button
            icon={ validateDni(props.identity) ? iconEnabled : iconDisabled  }
            onPress={ props.addFinalConsumer }
            buttonStyle={ style.buttonCheck }
            disabledStyle= { style.buttonCheckDisabled }
            loading = { props.loading }
            disabled = { props.loading || !validateDni(props.identity) }
          />
        </View>
      </View>
    );
  }
}

InvoiceCustomer.propTypes = {
  finalConsumer: PropTypes.bool,
  setFinalConsumer: PropTypes.func,
  addFinalConsumer: PropTypes.func,
  loading: PropTypes.bool,
  identity: PropTypes.string
};

export default InvoiceCustomer;