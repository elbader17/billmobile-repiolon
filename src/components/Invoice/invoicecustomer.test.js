import React from 'react';
import { shallow, mount } from 'enzyme';
import InvoiceCustomer from './InvoiceCustomer';

jest.mock('../../constants/metrics', () => ({ METRICS: { heightHeader: {} } }));

test('render correctly functional component Invoice', () => {
  const fcomponent =  shallow(<InvoiceCustomer />);
  expect(fcomponent).toMatchSnapshot();
});
