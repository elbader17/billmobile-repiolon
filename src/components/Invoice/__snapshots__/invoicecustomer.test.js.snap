// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly functional component Invoice 1`] = `
<ScrollViewMock>
  <View
    style={
      Object {
        "marginHorizontal": 8,
        "marginVertical": 5,
      }
    }
  >
    <InvoiceListCustomer />
  </View>
</ScrollViewMock>
`;
